File: 01-ordenacao/java/OrdenacaoMVC/src/model/GeradorLista.java
package model;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

/**
 * Gera listas de inteiros para entrada dos algoritmos
 */
public final class GeradorLista {

    private GeradorLista() {
    }

    /**
     * Método que cria uma lista preenchida com números aleatórios
     * @param quantidade números a gerar
     * @param minimo mínimo (inclusivo)
     * @param maximo máximo (exclusivo)
     */
    public static List<Integer> gerar(int quantidade, int minimo, int maximo) {
        List<Integer> lista = new ArrayList<>(quantidade);
        Random gerador = new Random();

        for (int i = 0; i < quantidade; i++) {
            lista.add(gerador.nextInt(minimo, maximo));
        }

        return lista;
    }
}

File: 01-ordenacao/csharp/OrdenacaoMVC/src/Model/GeradorLista.cs
namespace OrdenacaoMVC.Model;

/// <summary>
/// Gera listas de inteiros para uso como entrada dos algoritmos de ordenação
///</summary>
public static class GeradorLista
{
    /// <summary>
    /// Cria uma nova lista preenchida com números inteiros aleatórios
    ///</summary>
    /// <param name="quantidade">Quantidade de números a gerar</param>
    /// <param name="minimo">Valor mínimo (inclusivo) do intervalo</param>
    /// <param name="maximo">Valor máximo (exclusivo) do intervalo</param>
    public static List<int> Gerar(int quantidade, int minimo, int maximo)
    {
        List<int> lista = new(quantidade);
        Random gerador = Random.Shared;

        for (int i = 0; i < quantidade; i++)
        {
            lista.Add(gerador.Next(minimo, maximo));
        }

        return lista;
    }
}
